istioctl/pkg/writer/envoy/configdump/endpoint.go

```diff
@@ -67,21 +75,40 @@ func retrieveEndpointPort(ep *endpoint.LbEndpoint) uint32 {
 	return ep.GetEndpoint().GetAddress().GetSocketAddress().GetPortValue()
 }
 
-func retrieveEndpointAddress(ep *endpoint.LbEndpoint) string {
-	addr := ep.GetEndpoint().GetAddress()
-	if addr := addr.GetSocketAddress(); addr != nil {
-		return addr.Address + ":" + strconv.Itoa(int(addr.GetPortValue()))
+func retrieveEndpointAddresses(ep *endpoint.LbEndpoint) []string {
+	addrs := []*core.Address{ep.GetEndpoint().GetAddress()}
+	for _, a := range ep.GetEndpoint().GetAdditionalAddresses() {
+		addrs = append(addrs, a.GetAddress())
 	}
-	if addr := addr.GetPipe(); addr != nil {
-		return addr.GetPath()
+	result := make([]string, 0, len(addrs))
+	for _, addr := range addrs {
+		if addr := addr.GetSocketAddress(); addr != nil {
+			result = append(result, addr.Address+":"+strconv.Itoa(int(addr.GetPortValue())))
+			continue
+		}
+		if addr := addr.GetPipe(); addr != nil {
+			result = append(result, addr.GetPath())
+			continue
+		}
+		if internal := addr.GetEnvoyInternalAddress(); internal != nil {
+			switch an := internal.GetAddressNameSpecifier().(type) {
+			case *core.EnvoyInternalAddress_ServerListenerName:
+				result = append(result, fmt.Sprintf("envoy://%s/%s", an.ServerListenerName, internal.EndpointId))
+				continue
+			}
+		}
+		result = append(result, "unknown")
 	}
-	if internal := addr.GetEnvoyInternalAddress(); internal != nil {
-		switch an := internal.GetAddressNameSpecifier().(type) {
-		case *core.EnvoyInternalAddress_ServerListenerName:
-			return fmt.Sprintf("envoy://%s/%s", an.ServerListenerName, internal.EndpointId)
+
+	// Workaround https://github.com/envoyproxy/envoy/issues/35100. We can remove if it is fixed.
+	filtered := make([]string, 0, len(result))
```

Comment:
I will take care of removing the workaround after https://github.com/envoyproxy/envoy/pull/35120 is merged